server/src/main/java/org/elasticsearch/index/mapper/BlockSourceReader.java

```diff
@@ -149,6 +192,75 @@ public String toString() {
         }
     }
 
+    private static class Points extends BlockSourceReader {
+
+        Points(ValueFetcher fetcher) {
+            super(fetcher);
+        }
+
+        @Override
+        protected void append(BlockLoader.Builder builder, Object v) {
+            if (v instanceof SpatialPoint point) {
+                ((BlockLoader.PointBuilder) builder).appendPoint(point);
+            } else if (v instanceof String wkt) {
+                try {
+                    // TODO: figure out why this is not already happening in the GeoPointFieldMapper
+                    Geometry geometry = WellKnownText.fromWKT(GeometryValidator.NOOP, false, wkt);
+                    if (geometry instanceof Point point) {
+                        // TODO: perhaps we should not create points for later GC here, and pass in primitives only?
+                        ((BlockLoader.PointBuilder) builder).appendPoint(new SpatialPoint(point.getX(), point.getY()));
```

Comment:
Indeed, we should try not to create any more garbage than we have to here. The `_source` loading is going to allocate - it has to do so to get at `_source`, but we should avoid it when we can.

Reply:
We've simplified this now, by generating WKB in the value fetcher and not converting types.